### DISCUSSION TITLE
Filter record by JSONB array match

### DISCUSSION THREAD

**Discussion by stychu** [supabase/supabase#2387] [Questions]
Hello there.
Im trying to fetch single entry from my table that contains and JSONB array of objects. Can I match somehow that array to find the desired result?

```
[
   {
    "chats": [
      {
        "id":  56789,
      },
      {
        "id":  66753,
      },
    ],
    "id": 999
  },  
 {
    "chats": [
      {
        "id":  43532,
      }
    ],
    "id": 999
  }
]
  ```
  
  I would like to get the object that matches id `999` and contains in `chats` -> `id: 66753`
  
  Tried few approaches but none worked.

I though something link this will work.But no success  
  ```
  let { data, error } = await supabase
  .from('xyz')
  .select('*')

  .eq('id', 999)
  .contains('chats', {id: 66753})
  ```

Can it be done ?

**Comment by steve-chavez** [comment]
Hey @stychu,

The [contains](https://supabase.io/docs/reference/javascript/contains) filter is for arrays only, try it like:

```js
let { data, error } = await supabase
.from('xyz')
.select('*')

.eq('id', 999)
.contains('chats', [{id: 66753}]) //array used here
```

**Reply by stychu** [reply]
but this only work if my chats is the Object rather than array Right??

I have an array of object in my JSONB column and I get this

```
{ hint: null,
  details: 'Expected string or "}", but found "[".',
  code: '22P02',
  message: 'invalid input syntax for type json' }
  ```

**Reply by saendu** [reply]
Hey @stychu check my post about fuzzy search in arrays: https://github.com/supabase/supabase/discussions/2415#discussion-3464689 This should give you a hint what works and what not. For arrays however I also do not have a solution/answer yet.

**Reply by steve-chavez** [reply]
@stychu As mentioned by @saendu, does adding JSON.stringify work?

```js
let { data, error } = await supabase
.from('xyz')
.select('*')
.eq('id', 999)
.contains('chats', JSON.stringify([{id: 66753}]))
```
